### PR TITLE
docs: fix and extend build instructions

### PR DESCRIPTION
**Pull request by ingomueller-net**
This commit fixes several problems with the build instructions in the README and extends them. In particular, the following changes are made:

* List global prerequisites (like git, CMake, C++ compiler, etc.)
* Fix base path and define it as a first step
* Explicitly describe what paths to add to the `activate` script, including `PATH`, and adapt following commands to that
* Mention possible problems with using `lld`
* Remove the quirk with `-DABSL_ENABLE_INSTALL=ON` -- this doesn't seem to be necessary anymore
* Remove `lit` from the Python requirements because LLVM provides it in the `PATH`
* Adapt the `lit` examples to the new repository structure

### CODE REVIEW

**Review by mbrobbel — APPROVED** [review]

**Comment by ingomueller-net** [comment]
Thanks @mbrobbel for the review! (I still have to wait for a review from someone with write access...)

**Review by jpienaar — APPROVED** [review]

**Pull request merged**
